hadoop-yarn-project/hadoop-yarn/hadoop-yarn-api/src/main/java/org/apache/hadoop/yarn/server/api/protocolrecords/FederationQueueWeight.java

```diff
@@ -170,6 +170,26 @@ public static void checkHeadRoomAlphaValid(String headRoomAlpha) throws YarnExce
     }
   }
 
+  public static void checkPolicyManagerValid(String policyManager) throws YarnException {
+    switch (policyManager) {
+      case "org.apache.hadoop.yarn.server.federation.policies.manager.HashBroadcastPolicyManager":
```

Comment:
Can we use some of the getClass().getSimpleName() or similar methods?

Reply:
This is a good idea! I will improve this part of the code. I will add verification on the router side.